src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java

```diff
@@ -0,0 +1,233 @@
+        prdEnumServiceMock = mock(PrdEnumServiceImpl.class);
+        jurisdictionService = mock(JurisdictionServiceImpl.class);
+        prdEnumRepository = mock(PrdEnumRepository.class);
+        userProfileFeignClient = mock(UserProfileFeignClient.class);
+
+        organisation = new Organisation("Org-Name", OrganisationStatus.PENDING, "sra-id", "companyN", false, "www.org.com");
+        organisationResponse = new OrganisationResponse(organisation);
+        professionalUser = new ProfessionalUser("some-fname", "some-lname", "soMeone@somewhere.com", organisation);
+        SuperUser superUser = new SuperUser("some-fname", "some-lname", "some-email-address", organisation);
+        organisationEntityResponse = new OrganisationEntityResponse(organisation, false);
+
+
+        prdEnumList = new ArrayList<>();
+        prdEnumList.add(anEnum1);
+        prdEnumList.add(anEnum2);
+        prdEnumList.add(anEnum3);
+
+        jurisdEnumIds = new ArrayList<>();
+        jurisdEnumIds.add("Probate");
+        jurisdEnumIds.add("Bulk Scanning");
+        jurisdictions = new ArrayList<>();
+
+        Jurisdiction jurisdiction1 = new Jurisdiction();
+        jurisdiction1.setId("Probate");
+        Jurisdiction jurisdiction2 = new Jurisdiction();
+        jurisdiction2.setId("Bulk Scanning");
+        jurisdictions.add(jurisdiction1);
+        jurisdictions.add(jurisdiction2);
+
+        List<String> userRoles = new ArrayList<>();
+        userRoles.add("pui-user-manager");
+        newUserCreationRequest = new NewUserCreationRequest("some-name", "some-last-name", "some@email.com", userRoles, jurisdictions);
+        userCreationRequest = new UserCreationRequest("some-fname", "some-lname", "some@email.com", jurisdictions);
+        organisationCreationRequest = new OrganisationCreationRequest("test", "PENDING", "sra-id", "false", "number02", "company-url", userCreationRequest, null, null);
+        userProfileCreationRequest = new UserProfileCreationRequest("some@email.com", "some-name", "some-last-name", EN, PROFESSIONAL, EXTERNAL, userRoles);
+        response = Response.builder().status(200).reason("OK").body(mock(Response.Body.class)).request(mock(Request.class)).build();
+
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testCreateOrganisation() {
+        when(organisationServiceMock.createOrganisationFrom(organisationCreationRequest)).thenReturn(organisationResponse);
+        when(prdEnumServiceMock.getPrdEnumByEnumType(any())).thenReturn(jurisdEnumIds);
+        when(prdEnumRepository.findAll()).thenReturn(prdEnumList);
+
+        ResponseEntity<?> actual = organisationExternalController.createOrganisationUsingExternalController(organisationCreationRequest);
+
+        verify(organisationCreationRequestValidatorMock, times(1)).validate(any(OrganisationCreationRequest.class));
+        verify(organisationServiceMock, times(1)).createOrganisationFrom(organisationCreationRequest);
+
```

Comment:
verify repository check as-well

Reply:
This will be done in all classes where repository has been mocked and used

Reply:
This has been addressed in this commit: https://github.com/hmcts/rd-professional-api/pull/450/commits/2e6a2f8b1a7050bb52d904bd20a2c5638869a566